### PR TITLE
Initial userdev plugin implementation

### PR DESCRIPTION
**Pull request by jpenilla**
Implements a userdev plugin, which allows for building against Mojang-mapped Paper with sources attached in-IDE, with support for remapping compiled plugins to Spigot's runtime names.

Besides adding the userdev plugin, this PR also makes necessary changes to core and patcher to allow for Paper and forks to generate "paperweight development bundles", which are then published to a maven repository for the userdev plugin to consume as input.

See here for an example plugin using paperweight-userdev: https://github.com/jpenilla/paperweight-test-plugin

And here for the Paper changes: https://github.com/PaperMC/Paper/pull/6163

### CODE REVIEW

File: paperweight-lib/src/main/kotlin/tasks/RemapJar.kt
Comment (Proximyst) [inline paperweight-lib/src/main/kotlin/tasks/RemapJar.kt:38]: Perhaps a (private?) val instead

File: paperweight-lib/src/main/kotlin/tasks/RunForgeFlower.kt
Comment (Proximyst) [inline paperweight-lib/src/main/kotlin/tasks/RunForgeFlower.kt:41]: Val here too

**Review by Proximyst — APPROVED** [review]
Cool! I assume you've tested this, so a check from me.

**Comment by OxKitsune** [comment]
What's blocking this from being merged?

**Comment by electronicboy** [comment]
main person working on this project is prioritising their own health over rushing around to get this stuff pulled in

**Comment by OxKitsune** [comment]
Ah that's fair. Thanks for the update :)

**Review by DenWav — APPROVED** [review]
Overall looks good.

I mostly only did sanity checks and familiarizing myself with the code here, but I trust you and your code, and it all looked sane. Just a couple tiny things that popped up.

I don't want my lack of time and energy right now preventing this from moving forward, so after addressing those things, feel free to merge. Would be cool ify ou could clean up the commit history a bit before merging, and please rebase and merge when you do.

File: paperweight-lib/src/main/kotlin/util/utils.kt
Comment (jpenilla) [reply paperweight-lib/src/main/kotlin/util/utils.kt:45]: It is, the import + the toHex method using it were moved from GeneratePaperclipPatch. By converting the byte to an int first we can avoid the experimental function though.

**Comment by JRoy** [comment]
@limbo-app ready

**Comment by JRoy** [comment]
I tried @jpenilla

**Review by JRoy — DISMISSED** [review]

**Review by Proximyst — APPROVED** [review]
limbo-app 🍞y

**Comment by MattBDev** [comment]
Woo!!

**Pull request merged**
